.github/workflows/cicd.yml

```diff
@@ -0,0 +1,269 @@
+          echo "Testing image: ${IMAGE_TAG}"
+          
+          docker run -d --name test-container -p ${{ env.CONTAINER_PORT }}:${{ env.CONTAINER_PORT }} ${IMAGE_TAG}
+          
+          echo "Waiting for container to start..."
+          READY=false
+          for i in $(seq 1 ${{ env.CONTAINER_STARTUP_TIMEOUT }}); do
+            if docker ps | grep -q test-container; then
+              if curl -sf http://localhost:${{ env.CONTAINER_PORT }}/ > /dev/null 2>&1; then
+                echo "✅ Container is running and application is responding"
+                READY=true
+                break
+              fi
+            fi
+            sleep 1
+          done
+          
+          if [ "$READY" = false ]; then
+            echo "❌ Container is running but application not responding after ${{ env.CONTAINER_STARTUP_TIMEOUT }} seconds"
+            docker logs test-container
+            exit 1
+          fi
+          
+          if docker ps | grep -q test-container; then
+            echo "✅ Container started successfully"
+            docker logs test-container
+          else
+            echo "❌ Container failed to start or crashed"
+            docker logs test-container
+            exit 1
+          fi
+          
+          docker stop test-container
+          docker rm test-container
+
+  # Stage 5: Deploy to Azure Container Registry (main branch only)
+  deploy-to-acr:
+    name: Push to Azure Container Registry
+    runs-on: ubuntu-latest
+    needs: build-docker
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to Azure
+        uses: azure/login@v2
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: Build and push Docker image to ACR
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: true
+          tags: |
+            ${{ secrets.AZURE_ACR_LOGIN_SERVER }}/job-app-backend:${{ needs.build-docker.outputs.image-tag }}
+            ${{ secrets.AZURE_ACR_LOGIN_SERVER }}/job-app-backend:latest
+          cache-from: type=registry,ref=${{ secrets.AZURE_ACR_LOGIN_SERVER }}/job-app-backend:buildcache
+          cache-to: type=registry,ref=${{ secrets.AZURE_ACR_LOGIN_SERVER }}/job-app-backend:buildcache,mode=max
+
+      - name: Verify image in ACR
+        uses: azure/cli@v2
+        with:
+          azcliversion: latest
+          inlineScript: |
+            az acr repository show-tags --name $(echo ${{ secrets.AZURE_ACR_LOGIN_SERVER }} | cut -d'.' -f1) --repository job-app-backend
+
+  # Stage 6: Database Migrations & Seeding (main branch only)
+  database-setup:
+    name: Database Migrations & Seeding
+    runs-on: ubuntu-latest
+    needs: deploy-to-acr
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```

Comment:
Not sure we should have this here - I would remove the DB Migration altogether unless there's a specific need to have it here